File: covariance_blocks/basicCovBlock.m
classdef basicCovBlock < covarianceblock
    % 
    %
    % Author: Walter Livingston
    
    properties
    end

    methods
        function [S, reject] = calcInnovationCovarianceMatrix(obj, filter)
            H = filter.measurement_block.H;
            P = filter.P;
            R = filter.measurement_block.calcMeasurementCovarianceMatrix();
            z = filter.z;
            S = H*P*H' + R;
            obj.S = S;
            obj.r = sqrt(z'*(S^(-1))*z);

            reject = false;
        end
    end
end

File: examples/basicCovBlock.py
import numpy as np
from modular_kf import CovarianceBlock
from modular_kf.helpers import *

class BasicCovBlock(CovarianceBlock):
    def calcInnovationCovariance(self, filter) -> tuple[mat, bool]:
        H = filter.measurement_block.H
        P = filter.P
        R = filter.measurement_block.calcMeasurementCovarianceMatrix()
        z = filter.z
        self.S = np.asmatrix(H*P*H.T + R)
        self.r = np.sqrt(z.T*(self.S^(-1))*z)

        reject = False

        return (self.S, reject)